### ISSUE TITLE
Provide Agent registration via Events

### ISSUE THREAD

**Issue by mgazz** [OFMFWG/OFMF-Reference#15]
**Goal:** Allow Agents to register their presence by posting an Event to the OFMF. 
**Note:** A first example is available at the route `/EventListener` 
**Todo:**
- Update mechanism to support registration based on [latest discussions](https://github.com/OFMFWG/Docs/blob/main/draft/OFMF_server_and_Agent_interactions_based_on_events.docx) 
- Add support for all relevant events

**Comment by mgazz** [comment]
@mjaguil To test the current implementation we need a running Agent. A mock example can be found [here](https://gist.github.com/mgazz/5cfbb328267336dd00084b462e076325)

You can start the mock as following (install missing dependencies with `pip` if necessary):
```
python ./ofmf_agent_mock.py
```

We can test the mechanism following :
```
cat << EOF > newmanager.json
{
    "@odata.type": "#Event.v1_7_0.Event",
    "Id": "1",
    "Name": "Manager Created",
    "Context": "",
    "Events": [ {
      "EventType": "Other",
      "EventId": "4594",
      "Severity": "Ok",
      "Message": "New Manager Available at FQDN http://foo.bar.org and Port 1234 ",
      "MessageId": "Manager.1.0.ManagerCreated",
      "MessageArgs": [ "http://127.0.0.1", "1234" ],
      "OriginOfCondition": {
       "@odata.id": "/redfish/v1/Managers/Manager1"
      }}
    ]}
EOF
```

Start the OFMF
```
./setup.sh
```

Trigger the registration with an event example
```
curl -X POST \
               -H "Content-Type: application/json" \
               -d @newmanager.json http://127.0.0.1:5000/EventListener
```

Verify that the new Manager (`Manager1`) is now available
```
curl http://127.0.0.1:5000//redfish/v1/Managers
{
    "@odata.type": "#ManagerCollection.ManagerCollection",
    "Name": "Manager Collection",
    "Members@odata.count": 2,
    "Members": [
        {
            "@odata.id": "/redfish/v1/Managers/EBOF-SoC"
        },
        {
            "@odata.id": "/redfish/v1/Managers/Manager1"
        }
    ],
    "@odata.id": "/redfish/v1/Managers"
}
```

**Comment by christian-pinto** [comment]
This is done and implemented via #27
